### ISSUE TITLE
Validation set loss exploding for baseline realNVP training with CIFAR10

### ISSUE THREAD

**Issue by jichan3751** [PolinaKirichenko/flows_ood#5]
Hi,

I'm running baseline realNVP training with CIFAR10 with following command:
```
python3 train_unsup.py --dataset=CIFAR10 --data_path=DATA_PATH --logdir=LOG_DIR \
  --ckptdir=CKPTS_DIR --num_epochs=101 --lr=1e-4 --batch_size=32 --num_blocks=8 \
  --weight_decay=5e-5 --num_scales=3
```

I'm printing out loss and bits per dim (from `utils.bits_per_dim(x, loss_meter.avg)`) every epoch.
```
Epoch 0 train loss 10171.217 bpd 4.777 test loss 552956.313 bpd 259.683
Epoch 1 train loss 9057.872 bpd 4.254 test loss 70802.084 bpd 33.251
Epoch 2 train loss 8784.356 bpd 4.125 test loss 32934.562 bpd 15.467
Epoch 3 train loss 8596.157 bpd 4.037 test loss 225722.941 bpd 106.006
Epoch 4 train loss 8502.981 bpd 3.993 test loss 10015.814 bpd 4.704
Epoch 5 train loss 8409.448 bpd 3.949 test loss 28452.946 bpd 13.362
Epoch 6 train loss 8398.587 bpd 3.944 test loss 460854.291 bpd 216.430
Epoch 7 train loss 8294.218 bpd 3.895 test loss 132133586.144 bpd 62053.538
Epoch 8 train loss 8269.459 bpd 3.884 test loss 10145953.338 bpd 4764.817
Epoch 9 train loss 8202.209 bpd 3.852 test loss 1016279.574 bpd 477.273
Epoch 10 train loss 8155.723 bpd 3.830 test loss 228595.389 bpd 107.355
Epoch 11 train loss 8120.548 bpd 3.814 test loss 12699808.581 bpd 5964.177
```
The train stats seems to be fine but the test loss / bpd is diverging.
Could this be a bug?

The script with MNIST works fine, so maybe it is only the problem of CIFAR10.

**Comment by PolinaKirichenko** [comment]
Hi! I think lowering the learning rate (to at least 1e-5) and using Adam or AdamW optimizers with increased weight decay should improve training stability.
